web/packages/teleport/src/Account/ManageDevices/AddDevice/useAddDevice.ts

```diff
@@ -84,13 +84,23 @@ export default function useAddDevice(
     qrCode,
     auth2faType: cfg.getAuth2faType(),
     isPasswordlessEnabled: cfg.isPasswordlessEnabled(),
+    restrictDeviceUsage,
   };
 }
 
 export type State = ReturnType<typeof useAddDevice>;
 
 export type Props = {
   token: string;
+  /**
+   * Controls whether the user can customize whether the device should allow
+   * passwordless authentication. `null` means that the user gets to choose;
+   * other values mean that the component's call site decides what kind of
+   * device we're adding.
+   *
+   * TODO(bl-nero): Disallow `null` when cleaning up the old flow.
+   */
+  restrictDeviceUsage?: DeviceUsage;
```

Comment:
Nit: If we are assigning `null` to this field, we should define the type as:
```suggestion
  restrictDeviceUsage: DeviceUsage | null;
```

An optional property means that the property may either be `DeviceUsage` or it may be `undefined`.

Anyway, this doesn't do much difference for us since we don't have `strictNullChecks` enabled.

Reply:
You're right, correcting